packages/runtime/src/widgets/Form/Checkbox.tsx

```diff
@@ -53,6 +53,10 @@ export const CheckboxWidget: React.FC<CheckBoxProps> = (props) => {
 
   const formInstance = Form.useFormInstance();
 
+  useEffect(() => {
+    setChecked(values?.initialValue as boolean);
```

Comment:
Instead of casting, you can use the `Boolean` constructor which will actually convert it to a boolean.